src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -664,17 +672,20 @@ void MapSetLocationCmd(SPAWNINFO* pChar, char* szLine)
 		// FIXME: exception handling?
 		CVector3 pos{ std::stof(xloc), std::stof(yloc), std::stof(zloc) };
 
-		MakeMapLoc(params, label, tag, pos, isDefaultLocSettings);
+		MapLocTemplate* newMapLoc = new MapLocTemplate(params, label, tag, pos, isDefaultLocSettings);
```

Comment:
use unique_ptr instead of raw pointer for MapLocTemplate. Update vector to use std::unique_ptr and then you don't need to delete when removing. If you combine this with the other suggestions, then clearing the container or removing an item will do all the work for you of cleaning things up. also makes it virtually impossible to do something wrong.